doc/devel/gitwash/development_workflow.rst

```diff
@@ -172,6 +172,71 @@ When you are ready to ask for someone to review your code and consider a merge:
    pull request message.  This is still a good way of getting some preliminary
    code review.
 
+Update your pull request
+========================
+
+Often, you'll want to update your contribution based on the feedback in the
+pull request. While it's possible to simply push additional commits to your
+feature branch, this will clutter the history. Not every step in the
+development of a feature is worth to be recorded. In contrast, a clean
+history with one or a few thematically separted commits helps to keep it
+more understandable. Most small features will do with a single commit.
+
+There are two techniques to achieve this.
+
+Ammend an existing commit
+   Use this if you have don't need to keep track of the incremental changes
+   to your feature.
+Squash multiple commits
+   Use this to cleanup the history when you've already made multiple commits
+   to your feature branch.
+
+Both techniques rewrite the history of your feature branch. While generally
+one has to be careful with rewriting the history, this is intended and
+perfectly fine for updates to your feature branch.
```

Comment:
perhaps: "perfectly fine for commits that are currently on your feature branch only".

a point that should be clear is that commits already in master shouldn't be touched.